src/Builder/FrontendDialectTransformer.cpp

```diff
@@ -1323,20 +1327,50 @@ class FrontendGenImpl {
     int nOut = 0;
     getNodeInputs(node, inputs);
     nOut = node.output().size();
+    std::vector<Type> givenOutputTypes;
+
+    // We lack a way of specifying import behavior for custom domains. For now
+    // some are hard-coded here, but an extension specification would be
+    // preferred.
+    if (node.domain().compare("com.microsoft") == 0) {
+      Type outElementType = {};
+      if (opName == "DequantizeLinear") {
+        outElementType =
+            inputs.at(1).getType().cast<ShapedType>().getElementType();
+      } else if (opName == "QuantizeLinear") {
+        outElementType =
+            inputs.at(2).getType().cast<ShapedType>().getElementType();
+      }
+      if (outElementType) {
+        auto outElemTypeAttr = builder_.getNamedAttr(
+            "output_element_type", TypeAttr::get(outElementType));
+        attributes.push_back(outElemTypeAttr);
+        givenOutputTypes.push_back(outElementType);
+
+        auto shapeInferAttr = builder_.getNamedAttr(
+            "shape_infer_pattern", builder_.getStringAttr("MDBroadcast"));
+        attributes.push_back(shapeInferAttr);
+      }
+    }
+
     // ToFix: The type inference may go wrong if the element type of the output
     // of CustomOp is not the same as the first input.
-    buildOutputAndOperation<ONNXCustomOp>(node, inputs, nIn, nOut, attributes);
+    buildOutputAndOperation<ONNXCustomOp>(
+        node, inputs, nIn, nOut, attributes, givenOutputTypes);
   }
 
   void ImportNode(const onnx::NodeProto &node) {
-    std::string opName = node.op_type() + GetImportVersionOfNode(node);
-    auto handler = import_handler_map_.find(opName);
-    std::vector<std::string> funcs = options_.functionsToDecompose;
-    if (!(std::find(funcs.begin(), funcs.end(), opName) != funcs.end())) {
-      if (handler != import_handler_map_.end()) {
-        // It's a regular op with a registered handler.
-        (this->*(handler->second))(node);
-        return;
+    if (isDefaultDomain(node.domain()) || (node.domain() == "ai.onnx.ml") ||
+        (node.domain() == "ai.onnx.preview.training")) {
```

Comment:
How did you figure out this `ai.onnx.preview.training` domain? Will there be any others from onnx we are not considering?

Reply:
Would it make sense to filter for `ai.onnx` domains? Or that is too general? I mean something like:
```suggestion
        ("ai.onnx" in node.domain())) {
```
I used the example in `python` but I think you get the point 😄

Reply:
I'd prefer to explicitly follow the domains documented in the onnx releases we support.